### ISSUE TITLE
[REVIEW]: PWSpy: A Python library dedicated to the analysis of Partial Wave Spectroscopic Microscopy data.

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3957]
**Submitting author:** @nanthony21 (<a href="http://orcid.org/0000-0003-2882-2471">Nicholas Anthony</a>)
**Repository:** https://github.com/BackmanLab/PWSpy
**Branch with paper.md** (empty if default branch): 
**Version:** 1.0.0
**Editor:** @jgostick
**Reviewers:** @pr4deepr, @jgostick
**Archive:** 10.5281/zenodo.6381082

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962/status.svg)](https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962"><img src="https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962/status.svg)](https://joss.theoj.org/papers/8c2996669cee9e94c227f4427c545962)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/BackmanLab/PWSpy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@dxm447, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jgostick know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @jgostick
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/BackmanLab/PWSpy">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/BackmanLab/PWSpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nanthony21) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @pr4deepr
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/BackmanLab/PWSpy">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] #4072
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nanthony21) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @dxm447 it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3957 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.20 s (502.0 files/s, 56824.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          62           1615           3236           5425
MATLAB                           9             91            210            322
YAML                             7             29             21            165
JSON                             6              0              0             73
Markdown                         3             26              0             58
DOS Batch                        4             20              1             54
reStructuredText                 6             49             52             51
TOML                             1              2              1             13
Bourne Shell                     3              1              0             12
make                             1              4              6              9
-------------------------------------------------------------------------------
SUM:                           102           1837           3527           6182
-------------------------------------------------------------------------------


Statistical information for the repository '2b4013d031a2377ebcfb4419' was
gathered on 2021/11/29.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
4D-1-HPD-1                      1             2              1            0.00
Nick                            81          5435           2013            7.16
Nick Anthony                  1653         43308          25861           66.50
nanthony21                      69          3236           1730            4.77
nicholas.anthony@nor           148          5052          17339           21.53
nickAnthonyNU                    2            24             11            0.03

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Nick                       1886           34.7         17.3               44.27
Nick Anthony               8390           19.4         18.4               10.64
```

**Comment by jgostick** [comment]
@whedon add @pr4deepr as reviewer

**Comment by whedon** [comment]
OK, @pr4deepr is now a reviewer

**Comment by jgostick** [comment]
@whedon generate pdf from branch JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: :point_left:

**Comment by jgostick** [comment]
Hi @nanthony21, I have taken a look at your paper and it has a few shortcomings that you might want to address before the reviewers do.  It's a bit on the short side...I know that the joss paper is not meant to be a full length snooze-fest, but this is an opportunity for you to give an outline of how the code works, like which modules are offered, and maybe a few figures highlighting the output.  Also the statement of need really focuses on the ability to handle different file formats.  Could you not highlight the need in a more 'big picture' way, like consistent analyses between researchers, unit tested results for ensure accuracy, etc?

**Comment by nanthony21** [comment]
@jgostick , Thank you for the feedback. I will get started on adding more information to the paper.

**Comment by nanthony21** [comment]
@whedon generate pdf from branch JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: :point_left:

**Comment by nanthony21** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03957/joss.03957/10.21105.joss.03957.pdf) :page_facing_up: :point_left:

**Comment by jgostick** [comment]
@nanthony21 the new paper looks much nicer.  The reviewers will drop in to begin playing with the software, but be patient.  This process could take a few months given the holiday shut-down will slow things.

**Comment by jgostick** [comment]
Hi @nanthony21, I have just pinged both reviewers by email to remind them that this review is waiting.  Sorry for the delay.  The holidays and recent lock-downs have not doubt upended people's schedules.

**Comment by nanthony21** [comment]
Thank you, @jgostick . I'll stay posted for if there is anything I can do once reviews begin.

**Comment by pr4deepr** [comment]
@nanthony21 , Apologies for the delay. Just letting you know that I'm chipping away at the review.

**Comment by jgostick** [comment]
I have pinged both reviewers by email to see how things are progressing.  Sorry fro the delays.

**Comment by pr4deepr** [comment]
The software satisfies most of the criteria. 
* I am using the test data to verify the functionality using examples in the docs. I have been communicating with @nanthony21 on the [PWSpy repo](https://github.com/BackmanLab/PWSpy/issues). I am not as familiar with Partial Wave Spectroscopic microscopy, but as far as I can tell, most functionality claims are validated. Just waiting on testing a few more. 

* I am not sure about the Community contributions as I don't see `3) Seek support` within https://github.com/BackmanLab/PWSpy/blob/master/CONTRIBUTING.md . I may have misunderstood this point. @jgostick Should a contact be provided??

**Comment by pr4deepr** [comment]
I've completed my review and am happy with the software and prompt responses by @nanthony21.
Please note that I am not as familiar with partial wave spectroscopy microscopy data and any functionality claims have been tested only using the examples and datasets provided in the library.

I think PWSpy is modular with an intuitive API, and makes data preprocessing and analysis straightforward.  This is a significant step up from single-use scripts which may only work for images from a specific instrument. The ability to save analysis settings and software versions in PWSpy is quite useful as this is often overlooked and underreported in many publications. 

@nanthony21 has provided ample test data with example scripts and a detailed readme, so anyone can run this package.

**Comment by jgostick** [comment]
Thanks SOO much @pr4deepr.  The peer review process is such a thankless task but the whole enterprise grinds to a halt without it.

**Comment by pr4deepr** [comment]
Thank you for the opportunity. I enjoyed this a lot and learned quite a bit too..

**Comment by jgostick** [comment]
@editorialbot remove @dxm447 from reviewers

**Comment by jgostick** [comment]
Hi @nanthony21, having had no response from the other reviewer, I have removed them and just done the review myself.  Everything looks GREAT.  I was not able to test the 'functionality' thoroughly, since I don't really do this sort image analysis, but I do lots of other image analysis (tomograms of porous media) and I can clearly see that this package is first-rate.  So I will begin the process of accepting it!  Congrats and thanks for your patience.

**Comment by jgostick** [comment]
Hi @nanthony21, can you address the items in first checklist below?

# Post-Review Checklist for Editor and Authors
## Additional Author Tasks After Review is Complete
- [ ] Double check authors and affiliations (including ORCIDs)
- [ ] Make a release of the software with the latest changes from the review and post the version number here. This is the version that will be used in the JOSS paper.  
- [ ] Archive the release on Zenodo/figshare/etc and post the DOI here.
- [ ] Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

## Editor Tasks Prior to Acceptance
- [ ] Set article DOI with ``@editorialbot set <DOI here> as archive``
- [ ] Set version with ``@editorialbot set <version here> as version``
- [ ] Double check rendering of paper with ``@editorialbot generate pdf``
- [ ] Specifically check the references with ``@editorialbot check references`` and ask author(s) to update as needed
- [ ] Do a 'dry run' of acceptance with ``@editorialbot recommend accept``

**Comment by nanthony21** [comment]
- [x] `Double check authors and affiliations (including ORCIDs)`
  I've checked the author, affiliations, and ORCIDs and everything is correct.
 - [x] `Make a release of the software with the latest changes from the review and post the version number here. This is the version that will be used in the JOSS paper.`
 I've created release 1.0.0 which includes the changes resulting from @pr4deepr's review.
 - [x]  `Archive the release on Zenodo/figshare/etc and post the DOI here.`
 The repository has been archived on Zenodo. DOI is : [DOI: 10.5281/zenodo.6381082](https://doi.org/10.5281/zenodo.6381082)
 - [x] `Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.`

**Comment by nanthony21** [comment]
Thank you both very much for your time and feedback!

**Comment by jgostick** [comment]
@editorialbot set 1.0.0 as version

**Comment by jgostick** [comment]
@editorialbot set 10.5281/zenodo.6381082 as archive

**Comment by jgostick** [comment]
@editorialbot check references

**Comment by jgostick** [comment]
@editorialbot generate pdf

**Comment by jgostick** [comment]
@editorialbot add @jgostick as reviewer

**Comment by jgostick** [comment]
@editorialbot recommend accept

**Comment by jgostick** [comment]
@editorialbot commands

**Comment by jgostick** [comment]
@editorialbot recommend-accept

**Comment by jgostick** [comment]
Hi @nanthony21, thanks for you patience throughout this review, it was a bit slow.  An editor in chief should be along soon to accept it officially, well done!

**Comment by kyleniemeyer** [comment]
Hi @nanthony21, I made a small PR that fixed some of the references in the paper, can you merge this? https://github.com/BackmanLab/PWSpy/pull/21

I will publish your article after that.

**Comment by nanthony21** [comment]
Thanks @kyleniemeyer , the pull request has been merged.

**Comment by kyleniemeyer** [comment]
@editorialbot generate pdf

**Comment by kyleniemeyer** [comment]
@editorialbot accept

**Comment by kyleniemeyer** [comment]
Congratulations @nanthony21 on your article's publication in JOSS!

Many thanks to @pr4deepr and @jgostick for reviewing this, and @jgostick for editing.
